checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallInference.java

```diff
@@ -1,176 +1,873 @@
-   * #owningFields}.
+   * Analyze the arguments of the method invocation node. If any of them is passed as an owning
+   * parameter to the callee, and is an alias with any parameter of the current method, this method
+   * adds the {@code @Owning} annotation to the corresponding parameter of the current method.
    *
-   * @param curBlock the current block
-   * @param visited set of blocks already on the worklist
-   * @param worklist current worklist
+   * @param obligations the obligations associated with the current block
+   * @param invocation the method invocation node to check
    */
-  private void propagateRegPaths(Block curBlock, Set<Block> visited, Deque<Block> worklist) {
+  private void analyzeOwnershipTransferAtMethodCall(
+      Set<Obligation> obligations, MethodInvocationNode invocation) {
+    List<? extends VariableTree> paramsOfCurrentMethod = methodTree.getParameters();
+    if (paramsOfCurrentMethod.isEmpty()) {
+      return;
+    }
+    List<? extends VariableElement> invocationParams = mcca.getParametersOfInvocation(invocation);
+    if (invocationParams.isEmpty()) {
+      return;
+    }
+    List<Node> arguments = mcca.getArgumentsOfInvocation(invocation);
 
-    List<Block> successors = getNormalSuccessors(curBlock);
+    for (int i = 1; i < arguments.size() + 1; i++) {
+      if (!resourceLeakAtf.hasOwning(invocationParams.get(i - 1))) {
+        continue;
+      }
+      for (int j = 1; j < paramsOfCurrentMethod.size() + 1; j++) {
+        VariableTree paramOfCurrMethod = paramsOfCurrentMethod.get(j - 1);
+        if (resourceLeakAtf.hasEmptyMustCallValue(paramOfCurrMethod)) {
+          continue;
+        }
 
-    for (Block b : successors) {
-      // If b is a special block, it must be the regular exit, since we do not propagate to
-      // exceptional successors.
-      if (b.getType() == Block.BlockType.SPECIAL_BLOCK) {
-        WholeProgramInference wpi = typeFactory.getWholeProgramInference();
-        assert wpi != null : "MustCallInference is running without WPI.";
-        for (VariableElement fieldElt : owningFields) {
-          wpi.addFieldDeclarationAnnotation(fieldElt, OWNING);
+        Node arg = NodeUtils.removeCasts(arguments.get(i - 1));
+        VariableElement paramElt = TreeUtils.elementFromDeclaration(paramOfCurrMethod);
+        if (isParamAndArgAliased(obligations, arg, paramElt)) {
+          addOwningToParam(j);
+          break;
         }
       }
+    }
+  }
 
-      if (visited.add(b)) {
-        worklist.add(b);
+  /**
+   * Checks whether the given argument which is passed in the invocation node and the given
+   * parameter of the current method are resource aliases.
+   *
+   * @param obligations the obligations associated with the current block
+   * @param argument the argument
+   * @param paramElt the parameter
+   * @return true if the {@code paramElt} is in the resource alias set of the given {@code arg}
+   */
+  private boolean isParamAndArgAliased(
```

Comment:
Why is this method named "isParamAndArgAliased" rather than, say, "isNodeAndElementAliased"?  The method does not seem to rely on the fact that `argument` is an argument (and, what is it even an argument of?) or that `paramElt` is a parameter (again, of what?).  Describing the behavior of the method more precisely will make it easier to understand its specification and its implementation.  It won't harm understandability at where the method is invoked.